src/ComputeSharp.D2D1/Shaders/Interop/Factories/D2D1AffineTransformMapperFactory.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using ComputeSharp.D2D1.Interop;
+using ComputeSharp.D2D1.Shaders.Interop.Factories.Abstract;
+using ComputeSharp.D2D1.Shaders.Interop.Helpers;
+
+namespace ComputeSharp.D2D1.Shaders.Interop.Factories;
+
+/// <summary>
+/// A container for shader-specific affine transform mapper factory.
+/// </summary>
+/// <remarks>
+/// This non-generic type is used so that <see cref="Transform"/> and <see cref="SubtractAndClampToMaxIfOverflow"/> are not recompiled for each generic type instantiation.
+/// </remarks>
+internal sealed class D2D1AffineTransformMapperFactory
+{
+    /// <summary>
+    /// Transforms a target rectangle using a given transform matrix.
+    /// </summary>
+    /// <param name="matrix">The transform matrix to use.</param>
+    /// <param name="rectangle">The rectangle to transform.</param>
+    private static void Transform(in Matrix3x2 matrix, ref Rectangle64 rectangle)
+    {
+        // Get the rectangle corners
+        Point64 topLeft = new(rectangle.Left, rectangle.Top);
+        Point64 topRight = new(rectangle.Right, rectangle.Top);
+        Point64 bottomLeft = new(rectangle.Left, rectangle.Bottom);
+        Point64 bottomRight = new(rectangle.Right, rectangle.Bottom);
+
+        // Transform them with the current matrix
+        var transformedTopLeft = topLeft.Transform(in matrix);
+        var transformedTopRight = topRight.Transform(in matrix);
+        var transformedBottomLeft = bottomLeft.Transform(in matrix);
+        var transformedBottomRight = bottomRight.Transform(in matrix);
+
+        // Calculate the bounding box of the transformed points
+        double transformedLeft = Math.Min(Math.Min(transformedTopLeft.X, transformedTopRight.X), Math.Min(transformedBottomLeft.X, transformedBottomRight.X));
+        double transformedTop = Math.Min(Math.Min(transformedTopLeft.Y, transformedTopRight.Y), Math.Min(transformedBottomLeft.Y, transformedBottomRight.Y));
+        double transformedRight = Math.Max(Math.Max(transformedTopLeft.X, transformedTopRight.X), Math.Max(transformedBottomLeft.X, transformedBottomRight.X));
+        double transformedBottom = Math.Max(Math.Max(transformedTopLeft.Y, transformedTopRight.Y), Math.Max(transformedBottomLeft.Y, transformedBottomRight.Y));
+
+        // Round each coordinate as needed and compute the rectangle bounds
+        long left = (long)Math.Floor(Math.Max(transformedLeft, long.MinValue));
+        long top = (long)Math.Floor(Math.Max(transformedTop, long.MinValue));
+        long right = (long)Math.Ceiling(Math.Min(transformedRight, long.MaxValue));
+        long bottom = (long)Math.Ceiling(Math.Min(transformedBottom, long.MaxValue));
+
+        // Calculate the width and height, with clamping in case of overflows
+        long width = SubtractAndClampToMaxIfOverflow(right, left);
```

Comment:
alright, one more thing ... consider the case where we have `{ left = long.MinValue, top = long.MinValue, right = long.MaxValue, bottom = long.MaxValue }`. This code will cause us to have a rectangle at x=y=MinValue, width=height=MaxValue, which means that Right and Bottom will be equal to -1. 

Remember, 'logically infinite' for a RectInt64 is `{ x = long.MinValue / 2, y = long.MinValue / 2, width = long.MaxValue, height = long.MaxValue }`. So I think we should just clamp to that range instead (above, I mean, when calculating l/t/r/b)

(isn't this fun? 😂😭)

Reply:
This should already be handled, as discussed on Discord 🙂